src/main/java/duke/DukeException.java

```diff
@@ -0,0 +1,139 @@
+package duke;
+
+/**
+ * DukeException stores all duke-specific exceptions.
+ */
+public class DukeException extends Exception {
+
+    /** Exception type */
+    protected DukeExceptionType exceptionType;
+    /** Command type */
+    protected DukeCommandType commandType;
+
+    /**
+     * Initialises DukeException using exception type.
+     *
+     * @param e
+     * @param exceptionType
+     */
+    public DukeException(String e, DukeExceptionType exceptionType) {
+        super(e);
+        this.exceptionType = exceptionType;
+    }
+
+    /**
+     * Initialises DukeException using exception type and command type.
+     *
+     * @param e
+     * @param exceptionType
+     * @param commandType
+     */
+    public DukeException(String e, DukeExceptionType exceptionType, DukeCommandType commandType) {
+        super(e);
+        this.exceptionType = exceptionType;
+        this.commandType = commandType;
+    }
+
+    /**
+     * Returns error messages depending on the exception type and command type.
+     *
+     * @return error messages
+     */
+    @Override
+    public String toString() {
+        String error = "";
+        switch (exceptionType) {
+        case EMPTY_LIST:
+            error += "_____________________________________________________";
```

Comment:
Since error messages follow a similar format, maybe you could have another method e.g. createErrorMessage(String) to create the string?